apps/consulting/posts/review-functorch-contribution.md

```diff
@@ -0,0 +1,323 @@
+---
+title: 'Review: `torch.func` Contribution'
+published: September 22, 2023
+author: kshiteej-kalambarkar
+description: '`torch.func` (previously known as `functorch`) is a PyTorch module
+designed to offer JAX-like transforms. Within this module, various higher-order
+functions, such as `grad`, `vmap`, and `vjp` are made accessible. These
+transforms help users to easily compute gradients for the parameters of their model
+or write batch-size agnostic code. The beauty of these transformations lies in
+their ability to compose with one another. Thanks to this, the process of
+calculating per-sample gradients becomes the straightforward application of
+`vmap(grad(model))`.'
+---
+
+<base target="_blank" />
+
+`torch.func` (previously known as `functorch`) is a PyTorch module designed to
+offer JAX-like transforms. Within this module, various higher-order
+functions, such as `grad`, `vmap`, and `vjp` are made accessible. These
+transforms help users to easily compute gradients for the parameters of their model
+or write batch-size agnostic code. The beauty of these transformations lies in
+their ability to compose with one another. Thanks to this, the process of
+calculating per-sample gradients becomes the straightforward application of
+`vmap(grad(model))`.
+
+Here are a few of the tasks we had the opportunity to tackle:
+
+## Adding Batching Rules for `vmap`
```

Comment:
@kshitij12345 @lezcano What contributions did Quansight make here? The section doesn't say.

Reply:
Discussed over slack